modules/objdetect/src/cascadedetect.cpp

```diff
@@ -587,7 +587,7 @@ bool HaarEvaluator::read(const FileNode& node, Size _origWinSize)
     localSize = lbufSize = Size(0, 0);
     if (ocl::haveOpenCL())
     {
-        if (ocl::Device::getDefault().isAMD())
+        if (ocl::Device::getDefault().isAMD() || ocl::Device::getDefault().isIntel())
```

Comment:
@vpisarev Could you please provide the detail of your Intel platform. We could also do test on it.
Thanks.